### DISCUSSION TITLE
Possible to publish a ReadableStream?

### DISCUSSION THREAD

**Discussion by kevmo314** [webtorrent/webtorrent#2255] [Q&A]
Is it possible to publish a ReadableStream? This doesn't follow the BEP44 conventions, but what I'm looking for is using webtorrent to identify who has a data source then request data from them where the data isn't known at time of publication, only that the node can produce it.

Currently my best idea is to run the DHT separately and handle the actual data streaming on my own, but I'm also not clear on the best way to do that either.

**Comment by hello-smile6** [comment]
> Is it possible to publish a ReadableStream? This doesn't follow the BEP44 conventions, but what I'm looking for is using webtorrent to identify who has a data source then request data from them where the data isn't known at time of publication, only that the node can produce it.
> 
> Currently my best idea is to run the DHT separately and handle the actual data streaming on my own, but I'm also not clear on the best way to do that either.

Try using Bugout. You can always base64-encode your data and send it as text.
https://github.com/chr15m/bugout
